x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/TransportGetStackTracesAction.java

```diff
@@ -259,20 +327,18 @@ private ActionListener<SearchResponse> handleEventsGroupedByStackTrace(
         Client client,
         GetStackTracesResponseBuilder responseBuilder,
         ActionListener<GetStackTracesResponse> submitListener,
-        Function<SearchResponse, Map<String, Long>> stacktraceCollector
+        Function<SearchResponse, Map<String, TraceEvent>> stacktraceCollector
     ) {
         StopWatch watch = new StopWatch("eventsGroupedByStackTrace");
         return ActionListener.wrap(searchResponse -> {
-            Min minTimeAgg = searchResponse.getAggregations().get("min_time");
-            Max maxTimeAgg = searchResponse.getAggregations().get("max_time");
-            long minTime = Math.round(minTimeAgg.value());
-            long maxTime = Math.round(maxTimeAgg.value());
+            long minTime = getAggValueAsLong(searchResponse, "min_time");
+            long maxTime = getAggValueAsLong(searchResponse, "max_time");
 
-            Map<String, Long> stackTraceEvents = stacktraceCollector.apply(searchResponse);
+            Map<String, TraceEvent> stackTraceEvents = stacktraceCollector.apply(searchResponse);
 
             log.debug(watch::report);
             if (stackTraceEvents.isEmpty() == false) {
-                long totalSamples = stackTraceEvents.values().stream().mapToLong(v -> v).sum();
+                long totalSamples = stackTraceEvents.values().stream().mapToLong(v -> v.count).sum();
```

Comment:
This is obsolete now as we call `setTotalSamples` from both `searchProfilingEvents` and `searchGenericEvents`.